### DISCUSSION TITLE
How to build a run-anywhere shared object (.so) ?

### DISCUSSION THREAD

**Discussion by JaydenFish** [jart/cosmopolitan#1249] [Ideas]
run in any linux os is an acceptable idea.

**Comment by G4Vi** [comment]
It's been considered, but is not a priority for us. If we were to implement it we'd wouldn't be okay with just support Linux. Non-cosmo share libraries can be loaded with `cosmo_dlopen`.

jart said
> In order for that to be possible, someone would need to pick a file format (e.g. so, dylib, dll) and write a cross-platform dynamic loader for it. To date, no one's done that yet. Cosmopolitan has only so far focused on making executables portable. However Cosmopolitan does provide a cosmo_dlopen() function for loading dynamic libraries provided by your distro/os.

https://github.com/jart/cosmopolitan/issues/1081

Edit: before linked wrong issue
